### PR TITLE
Configurable encryption

### PR DESCRIPTION
**Pull request by danfinlay**
Allows custom encryptor to be passed to MetaMaskController and KeyringController.

Adds documentation of the `platform` object to the "developing for a new platform" guide.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -95,6 +95,7 @@ module.exports = class MetamaskController extends EventEmitter {
       initState: initState.KeyringController,
       ethStore: this.ethStore,
       getNetwork: this.networkController.getNetworkState.bind(this.networkController),
+      encryptor: opts.platform.encryptor || undefined,
```
Comment (kumavis) [inline app/scripts/metamask-controller.js]: the platform obj is intended to perform the same operation on different platforms

the encryptor should be configurable independent platform (e.g. cryptoX in electron, cryptoY in electron)

**Review by kumavis — CHANGES_REQUESTED** [review]

File: app/scripts/metamask-controller.js (hunk 1 above)
Comment (flyswatter) [reply app/scripts/metamask-controller.js]: That's why I've made it configurable here. Is there a problem you see with this implementation?

Comment (flyswatter) [reply app/scripts/metamask-controller.js]: In the docs it describes the required API for the encryptor object.

Comment (kumavis) [reply app/scripts/metamask-controller.js]: the encryptor should not be inside the platform obj

**Comment by flyswatter** [comment]
Made requested changes.

**Review by kumavis — APPROVED** [review]
beaut

**Comment by gmcdev** [comment]
Where this has been a blocker for me is:
https://github.com/danfinlay/browser-passworder/blob/master/index.js#L82-L101

Does this PR affect that dependency's execution in Metamask?

**Comment by danfinlay** [comment]
Yes. If you read the doc associated with the PR, providing your own passworder module will use it instead of the default one.

**Comment by gmcdev** [comment]
Yah, I read https://github.com/MetaMask/metamask-extension/blob/master/docs/porting_to_new_environment.md -- it wasn't immediately clear to me that the encryptor replaces `browser-passworder`, but now that you said it, it seems obvious. Thank you~

**Pull request merged**